src/routes/member/[studentId]/profile-picture/+page.server.ts

```diff
@@ -0,0 +1,175 @@
+    `${params.studentId}/profile-picture`,
+    true
+  );
+  return {
+    member,
+    photos,
+  };
+};
+
+export const actions = {
+  change: async ({ params, locals, request }) => {
+    const session = await locals.getSession();
+    const studentId = params.studentId;
+    await ctxAccessGuard(apiNames.MEMBER.UPDATE, session?.user, { studentId });
+    const formData = await request.formData();
+    const url = (formData.get("url") as string | null) ?? undefined;
+    if (!url || typeof url !== "string")
+      return fail(400, {
+        error: "Invalid url",
+        data: Object.fromEntries(formData),
+      });
+    try {
+      await prisma.member.update({
+        where: { studentId },
+        data: {
+          picturePath: url,
+        },
+      });
+      return {
+        success: true,
+        data: Object.fromEntries(formData),
+      };
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === "P2025" || e.code === "2016") {
+          return fail(404, { error: "Member not found", data: Object.fromEntries(formData) });
+        }
+        return fail(500, {
+          error: e.message ?? "Unknown error",
+          data: Object.fromEntries(formData),
+        });
+      }
+    }
+  },
+  upload: async ({ params, locals, request }) => {
+    const session = await locals.getSession();
+    const studentId = params.studentId;
+    await ctxAccessGuard(apiNames.MEMBER.UPDATE, session?.user, { studentId });
+    const failWithData = (error: string, statusCode: number = 400) => {
+      // in case of an error return the data and errors
+      formData.delete("image"); // not serializable
+      const data = {
+        data: Object.fromEntries(formData),
+        error,
+      };
+      return fail(statusCode, data);
+    };
+    const formData = await request.formData();
+    const image = formData.get("image") as File | null;
+    if (!image || !(image instanceof File) || image.size <= 0) return failWithData("Invalid file");
+    const fileName = generateUUID();
+    // get x,y,width,height from front end
+    const width = formData.get("crop-width") as string | null;
+    const height = formData.get("crop-height") as string | null;
+    const x = formData.get("crop-x") as string | null;
+    const y = formData.get("crop-y") as string | null;
+    if ((width && height && width !== height) || Number.isNaN(Number(width)) || Number(width) <= 0)
+      return failWithData("Invalid crop size");
+    try {
+      const buffer = await sharp(await image.arrayBuffer())
+        // this is required to keep the image upright
+        .rotate()
+        // crop image according to frontend settings
+        .extract({
+          left: Number(x ?? "0"),
+          top: Number(y ?? "0"),
+          width: Number(width),
+          height: Number(height),
+        })
+        // resize to MAX 400x400
+        .resize(400, 400, {
+          fit: "cover",
+          withoutEnlargement: true,
```

Comment:
Maybe image should be enlarged?

Reply:
no I don't think so. It will just take more space on the server without any more quality. The frontend will scale it anyway